compiler/src/iree/compiler/Codegen/Utils/GPUUtils.h

```diff
@@ -101,6 +102,16 @@ Value packVectorToSupportedWidth(Location loc, OpBuilder &builder, Value input);
 Value unpackToVector(Location loc, OpBuilder &builder, Value packedInput,
                      VectorType targetVecType);
 
+// A `dealloc` is converted into a call to `free` on the underlying data buffer.
+// The memref descriptor being an SSA value, there is no need to clean it up
+// in any way.
+struct DropSharedMemoryDeallocOp : public OpRewritePattern<memref::DeallocOp> {
```

Comment:
THe norm is to not have the pattern exposed directly this way. Could you move these patterns to `Common/GPU` and use a `populateDropSharedMemoryDeallocPatterns` method to add these patterns to both paths?

Reply:
I moved it.